src/sisl/sparse_geometry.py

```diff
@@ -3006,6 +3006,108 @@ def assert_unique(old, new):
         out._csr = csr
         out._geometry = geom
         return out
+    
+    def translate2uc(self):
+        """Translates all primary atoms to the unit cell.
+        
+        With this, the coordinates of the geometry are translated to the unit cell
+        and the supercell connections in the matrix are updated accordingly.
+
+        Returns
+        --------
+        SparseOrbital
+            A new SparseOrbital with the updated connections and a new associated geometry.
+        """
+        # Get the fractional coordinates of the associated geometry
+        geom = self.geometry
+        fxyz = geom.fxyz
+
+        # Get the quotient and remainder of dividing by 1.
+        # The quotient is the supercell index, the remainder is the coordinates in the unit cell.
+        current_sc, uc_fcoords = np.divmod(fxyz, 1)
+        
+        # Simply translate the atoms to move all atoms to the unit cell. That is, all atoms
+        # should be moved to supercell (0,0,0).
+        return self.translate_atoms_sc(-current_sc, geometry_coords=uc_fcoords @ self.cell)
+
+    def translate_atoms_sc(self, sc_translations, geometry_coords=None):
```

Comment:
Hmm... I don't particularly like this method. Couldn't we start with the `translate2uc` with the atoms specification, and if there is a need for the `translate_atoms_sc` then reconsider?

I see this method as being the equivalent of a `translate` method, since the `geometry_coords` can change the coords at will.

Reply:
I will remove the `geometry_coords` argument, it was just to avoid recomputing the coordinates in the particular case of `translate2uc` but now I see that it's exactly the same amount of computation whether I precompute or compute inside the method.

I can also make the method private for now. But I thought it was a nice separation of functionality since the only specificity of `translate2uc` is to know how much unit cells is each atom translated. And since the translation of the matrix elements is not trivial to implement I figured it would be nice to keep it like this.